### PR TITLE
Feature/advanced search appointments

### PR DESCRIPTION
**Pull request by Jana211**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 1]
```diff
@@ -221,6 +221,95 @@ private Boolean CheckIfAppointment(String stringToCheck)
...
+        /// <param name="appointments"> List of all <c>DoctorAppointment</c> of logged user.
+        /// </param>
+        /// <param name="dto"><c>AppointmentAdvancedSearchDto</c> is Data Transfer Object of a <c>DoctorAppointment</c> that is be used to filter appointments.
+        /// </param>
+        /// <param name="firstAppointments"> List of <c>DoctorAppointment</c> that contains appointments that matches first parameter.
+        /// </param>
+        /// <returns> List of filtered appointments. </returns>
+        private List<DoctorAppointment> searchForOtherParameters(List<DoctorAppointment> appointments, AppointmentAdvancedSearchDto dto, List<DoctorAppointment> firstAppointments)
+        {
+            List<DoctorAppointment> finalAppointments = firstAppointments;
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:245]: Remove line 245: unnecessary List<DoctorAppointment> finalAppointments

**Review by velicko-tam — CHANGES_REQUESTED** [review]
PR is not approved. Fix marked parts.

**Review by velicko-tam — APPROVED** [review]
PR is approved. Good Clean code principles were implemented.

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 2]
```diff
@@ -221,6 +221,93 @@ private Boolean CheckIfAppointment(String stringToCheck)
...
+        /// <param name="firstAppointments"> List of <c>DoctorAppointment</c> that contains appointments that matches first parameter.
+        /// </param>
+        /// <returns> List of filtered appointments. </returns>
+        private List<DoctorAppointment> searchForOtherParameters(List<DoctorAppointment> appointments, AppointmentAdvancedSearchDto dto, List<DoctorAppointment> firstAppointments)
+        {
+            for (int i = 0; i < dto.RestRoles.Length; i++)
+            {
+                List<DoctorAppointment> othersAppointments = searchForOtherRoles(dto.RestRoles[i], dto.Rest[i], appointments);
+                
+                firstAppointments = searchForLogicOperators(dto.LogicOperators[i], othersAppointments, firstAppointments);
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:249]: Write this in the form:
firstAppointments = searchForLogicOperators(dto.LogicOperators[i], searchForOtherRoles(dto.RestRoles[i], dto.Rest[i], appointments), firstAppointments);

**Review by trki293 — CHANGES_REQUESTED** [review]
Correction required in code.

**Review by trki293 — APPROVED** [review]
PR is approved. Clean code is followed.

**Pull request merged**